File: rebus-anagraficasrv/src/java/it/csi/rebus/anagraficasrv/business/service/EnteService.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2022 Regione Piemonte
* SPDX-License-Identifier: EUPL-1.2
*/
package it.csi.rebus.anagraficasrv.business.service;

import java.util.List;

import it.csi.rebus.anagraficasrv.vo.EnteVO;
import it.csi.rebus.anagraficasrv.vo.TipoEnteVO;

public interface EnteService {

	List<TipoEnteVO> getTipiEnte();

	List<EnteVO> getEnti();

	List<TipoRaggruppamentoVO> getTipiRaggruppamento();
}

File: rebus-anagraficawcl/src/app/services/ente.service.ts
/*************************************************
Copyright Regione Piemonte - 2022
SPDX-License-Identifier: EUPL-1.2-or-later
***************************************************/
import { Injectable } from "@angular/core";
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TipoEnteVO } from '../vo/tipo-ente-vo';
import { TipoRaggruppamentoVO } from '../vo/drop-down-menu-vo';
import { EnteVO } from '../vo/ente-vo';

@Injectable()
export class EnteService {


    urlGlobal: string = "/anagrafichetplweb/restfacade/ente";

    constructor(
        private http: HttpClient,
    ) { }

    //------------------------- GET ------------------------------------
    getTipiEnte(): Observable<Array<TipoEnteVO>> {
        let url: string = this.urlGlobal + '/tipiEnte';
        return this.http.get<Array<TipoEnteVO>>(url);
    }

    getEnti(): Observable<Array<EnteVO>> {
        let url: string = this.urlGlobal + '/enti';
        return this.http.get<Array<EnteVO>>(url);
    }

    getTipiRaggruppamento(): Observable<Array<TipoRaggruppamentoVO>> {
        let url: string = this.urlGlobal + '/tipiRaggruppamento';
        return this.http.get<Array<TipoRaggruppamentoVO>>(url);
    }

}